### PR TITLE
8308046: Move Solaris related charsets from java.base to jdk.charsets module

### PR DESCRIPTION
**Pull request by takiguc**
According to "JDK 20 Internationalization Guide"
https://docs.oracle.com/en/java/javase/20/intl/supported-encodings.html
Following Solaris related charsets are in "contained in jdk.charsets module" list.

- PCK (x-PCK)
- EUC_JP_Solaris (x-eucJP-Open)
- Big5_Solaris (x-Big5-Solaris)

These are not supported by Linux platform, so they should not be in java.base module.

Note:
GHA Linux x86 builds were failed.
I think it's not related by my modified code.
I opened [JDK-8308051](https://bugs.openjdk.org/browse/JDK-8308051) GHA: Linux x86 builds failure

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8308046](https://bugs.openjdk.org/browse/JDK-8308046): Move Solaris related charsets from java.base to jdk.charsets module

### Reviewers
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13973/head:pull/13973` \
`$ git checkout pull/13973`

Update a local copy of the PR: \
`$ git checkout pull/13973` \
`$ git pull https://git.openjdk.org/jdk.git pull/13973/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13973`

View PR using the GUI difftool: \
`$ git pr show -t 13973`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13973.diff">https://git.openjdk.org/jdk/pull/13973.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13973#issuecomment-1547118832)

### CODE REVIEW

**Comment by naotoj** [comment]
Hi @takiguc,
I'd expect this change not specific to Japanese, but to all such as `Big-5`. BTW, I am thinking of removing Charsets for Solaris ([8308117](https://bugs.openjdk.org/browse/JDK-8308117))

**Comment by naotoj** [comment]
I now think it is better simply removing Solaris-related charsets, as moving them from java.base to jdk.charsets would require unnecessary code changes in non-Solaris code.

**Comment by takiguc** [comment]
Hello @naotoj .
I'm not sure we can remove Solaris related charsets.
Somebody may use them for text communication between Solaris.
The latest change can move Big5_Solaris from java.base to jdk.charsets module.

**Comment by naotoj** [comment]
> Hello @naotoj . I'm not sure we can remove Solaris related charsets. Somebody may use them for text communication between Solaris.

OK, maybe not now.

I think the fix may be simplified by changing access for those `DoubleByte-X.java.template` internals, such as `DecodeHolder` to `public`, instead of introducing those access methods. You can import classes in `java.base/sun.nio.cs` with the wild card so that it would work on all platforms (`Big5` either in `java.base` or `jdk.charsets`)

Also, please drop `Japanese` from the issue/PR title

File: src/jdk.charsets/share/classes/sun/nio/cs/ext/EUC_JP_Open.java.template [hunk 1]
```diff
@@ -63,6 +63,12 @@ public class EUC_JP_Open
     }
 
     private static class Decoder extends EUC_JP.Decoder {
+        private static final SingleByte.Decoder DEC0201 =
+            (SingleByte.Decoder)new JIS_X_0201().newDecoder();
+
+        private static final DoubleByte.Decoder DEC0208 =
+            (DoubleByte.Decoder)new JIS_X_0208().newDecoder();
+
```
Comment (naotoj) [inline src/jdk.charsets/share/classes/sun/nio/cs/ext/EUC_JP_Open.java.template:71]: Can these also be removed, by making those original ones in `EUC_JP` public?

**Comment by naotoj** [comment]
>You can import classes in `java.base/sun.nio.cs` with the wild card so that it would work on all platforms (`Big5` either in `java.base` or `jdk.charsets`)

Scratch that, you've already did it. Then you can remove these: 
```
import sun.nio.cs.DoubleByte;
import sun.nio.cs.HistoricallyNamedCharset;
```

**Comment by takiguc** [comment]
Hello @naotoj .
I'd like to confirm about DoubleByte-X.java.template and EUC_JP.java.template.
I think the values are package-private.
Even if class is changed to `public`, the classes in`sun.nio.cs.ext` package could not access to these values in `sun.nio.cs` package...
I may be misunderstanding your suggestion, could you tell me more ?

**Comment by naotoj** [comment]
> I think the values are package-private. Even if class is changed to `public`, the classes in`sun.nio.cs.ext` package could not access to these values in `sun.nio.cs` package... 

I meant making those package-private fields public. I believe it's OK because java.base/sun.nio.cs package is only exported to jdk.charsets module.

**Comment by takiguc** [comment]
Thanks @naotoj .
I changed related fields to `public`.

**Comment by takiguc** [comment]
Hello @naotoj .
I appreciate your attention about JBS side.
I changed title and description, add noreg-cleanup label.

**Pull request closed without merging**